packages/babel-traverse/src/scope/index.js

```diff
@@ -131,14 +139,6 @@ const collectorVisitor = {
     scope.getBlockParent().registerDeclaration(path);
   },
 
-  ClassDeclaration(path) {
-    const id = path.node.id;
-    if (!id) return;
-
-    const name = id.name;
-    path.scope.bindings[name] = path.scope.getBinding(name);
```

Comment:
Isn't the following change all that's necessary?

```diff
-    path.scope.bindings[name] = path.scope.getBinding(name);
+    path.scope.bindings[name] = path.scope.parent.getBinding(name);
```

Reply:
I agree. `parent` is the nearest function scope, but classes are block-scoped:
```js
let A;
function fn() {
  {
    class A {}
  }
  A; // This references the let declaration, not the path.
}
```

Reply:
This change
```diff
-    path.scope.bindings[name] = path.scope.getBinding(name);
+    path.scope.bindings[name] = path.scope.parent.getBinding(name);
```
implicitly depends on the fact that the `Declaration` visitor runs _before_ `ClassDeclaration` visitor, so that `path.scope.parent.getBinding` always points to the new binding which may be initialized in `Declaration` visitor or `BlockScoped` visitor.

See https://github.com/babel/babel/issues/10896#issuecomment-574415561

Besides that, I agree we should do `path.scope.parent` instead of `parent`.

Reply:
@regiontog Ah you are right! The `ClassDeclaration` binding is registered in `BlockScoped` and `BlockScoped` runs _before_ `ClassDeclaration`. I am curious why the test can even get passed.